packages/renderer/src/lib/ui/DropdownMenu.svelte

```diff
@@ -0,0 +1,57 @@
+<script lang="ts">
+import { faEllipsisVertical } from '@fortawesome/free-solid-svg-icons';
+import Fa from 'svelte-fa/src/fa.svelte';
+export let backgroundColor = 'bg-zinc-800';
+export let enabled: boolean = true;
+
+// Show and hide the menu using clickOutside
+let showMenu = false;
+
+// If we touch outside the window, hide the menu
+let outsideWindow;
+
+// If we hit ESC while the menu is open, close it
+function handleEscape({ key }) {
+  if (key === 'Escape') {
+    showMenu = false;
+  }
+}
+
+function toggleMenu() {
+  showMenu = !showMenu;
+}
+
+// If we click outside the menu, close the menu
+function onWindowClick(e) {
+  if (outsideWindow.contains(e.target) == false) showMenu = false;
+}
+</script>
+
+<!-- Required in order for Svelte to track all key presses and if you pressed "ESC" -->
+<svelte:window on:keyup="{handleEscape}" on:click="{onWindowClick}" />
+
+<!-- If enabled, we'll show it, otherwise, just output the buttons normally.-->
+{#if enabled}
+  <!-- Create a "kebab" menu for additional actions. -->
+  <div class="relative inline-block text-left" bind:this="{outsideWindow}">
+    <!-- Button for the dropdown menu -->
+    <button
+      on:click="{() => {
+        toggleMenu();
+      }}"
+      class="mx-1 px-3 py-2 hover:text-violet-600 font-medium rounded-lg text-sm inline-flex items-center text-center {backgroundColor}">
+      <Fa class="h-4 w-4 text-xl" icon="{faEllipsisVertical}" />
+    </button>
+
+    <!-- Dropdown menu for all other actions -->
+    {#if showMenu}
+      <div
+        id="dropdown"
+        class="origin-top-right absolute right-0 z-10 m-2 rounded-md shadow-lg bg-zinc-700 ring-1 ring-black ring-opacity-5 divide-y divide-gray-600 focus:outline-none">
+        <slot />
+      </div>
+    {/if}
```

Comment:
really good improvement over the previous version
esc key works, display is not collapsed on the right

https://user-images.githubusercontent.com/436777/202119629-9cfb9f6c-2149-4d66-98d4-98f147ba6b6f.mp4

but the drop down should be displayed before the pointer when we click on elements at the bottom because while it works fine for upper rows, clicking on the rows near the bottom, the menu is displayed at the bottom and then we need another scroll. It should be displayed in that case at the opposite side

![image](https://user-images.githubusercontent.com/436777/202120474-6187656d-1882-4733-a82e-17e072013e82.png)